compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -1854,6 +1854,140 @@ TR::Register *OMR::X86::TreeEvaluator::csubEvaluator(TR::Node *node, TR::CodeGen
    return targetRegister;
    }
 
+TR::Register *OMR::X86::TreeEvaluator::bmulEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Register        *targetRegister       = NULL;
+   TR::Node            *firstChild           = node->getFirstChild();
+   TR::Node            *secondChild          = node->getSecondChild();
+   bool                 countsAreDecremented = false;
+   TR::MemoryReference *tempMR               = NULL;
+   bool                 oursIsTheOnlyMemRef  = true;
+
+   bool isMemOp = node->isDirectMemoryUpdate();
+   if (isMemOp)
```

Comment:
`isDirectMemoryUpdate` is used for operations that can directly update the memory, like `ADD r/m64, imm32` or `SUB r/m64, imm32`. It's a kind of optimization where we can avoid using an extra register when we can update the memory directly.  However, `imul` 's result always have to be stored into some part of RAX, so we can't do this optimization for `imul`.